xdsl/dialect_interfaces.py

```diff
@@ -0,0 +1,70 @@
+class DialectInterface:
+    """
+    A base class for dialects' interfaces.
+    They usually define functionality which is dialect specific to some transformation.
+
+    For example DialectInlinerInterface defines which dialect operations can be inlined and how.
+    Dialects will implement this interface and the inlining transformation will query them through the base interface.
+
+    The design logic tries to follow MLIR's dialect interfaces closely
+    https://mlir.llvm.org/docs/Interfaces/#dialect-interfaces
+    """
+
+    pass
+
+
+class OpAsmDialectInterface(DialectInterface):
```

Comment:
It would be nice to add a doc string to make this page look a bit better:
https://xdsl--4503.org.readthedocs.build/4503/reference/dialect_interfaces/index.html#xdsl.dialect_interfaces.OpAsmDialectInterface.lookup

You can navigate to the updated pages from any PR from the readthedocs action result.